builtin/mainmenu/tab_online.lua

```diff
@@ -264,19 +265,49 @@ end
 
 --------------------------------------------------------------------------------
 
+local function parse_search_input(input)
+	-- Return nil if nothing to search for
+	if not input:find("%S") then
+		return nil
+	end
+
+	local query = {keywords = {}, mods = {}, players = {}}
+
+	-- Process quotation enclosed parts
+	input = input:gsub("([^ ]?)\"([^\"]*)\"([^ ]?)", function(before, match, after)
+		if #before == 0 and #after == 0 then -- Also have be separated by spaces
+			table.insert(query.keywords, match)
+			return " "
+		end
+		return before..'"'..match..'"'..after
+	end)
+
+	-- Separate by space characters and handle special prefixes
+	-- (words with special prefixes need an exact match and none of them can contain spaces)
+	for word in input:gmatch("%S+") do
+		local mod = word:match("^mod:(.*)")
+		table.insert(query.mods, mod)
+		local player = word:match("^player:(.*)")
+		table.insert(query.players, player)
+		local game = word:match("^game:(.*)")
+		query.game = game
```

Comment:
Important: this must be

```suggestion
		query.game = query.game or game
```
otherwise the first non-game search term sets `query.game` to `nil`.